packages/smooth_app/lib/pages/product_list_user_dialog_helper.dart

```diff
@@ -220,163 +239,59 @@ class _UserListsDialogContent extends StatefulWidget {
 }
 
 class _UserListsDialogContentState extends State<_UserListsDialogContent> {
-  final List<String> all = <String>[];
-  final List<String> withBarcode = <String>[];
-  final Set<String> newWithBarcode = <String>{};
-
-  late bool _isLoading;
-
-  @override
-  void initState() {
-    super.initState();
-    _loadUserLists();
-  }
-
-  Future<void> _loadUserLists() async {
-    _isLoading = true;
-
-    await _loadAllUserLists();
-
-    withBarcode.clear();
-    withBarcode.addAll(await widget.daoProductList
-        .getUserLists(withBarcode: widget.product.barcode));
-
-    newWithBarcode.clear();
-    newWithBarcode.addAll(withBarcode);
-
-    if (mounted) {
-      _isLoading = false;
-      setState(() {});
-    }
-  }
-
-  Future<void> _loadAllUserLists() async {
-    all.clear();
-    all.addAll(await widget.daoProductList.getUserLists());
-  }
+  final Set<String> addTo = <String>{};
+  final Set<String> removeFrom = <String>{};
```

Comment:
I cannot see the point of those lists, and they should be private anyway.

Reply:
Private ✅ 
Added comments ✅ 

They are to keep track of the changes the user clicks in the list dialog. 

The barcode was in these lists

But he wants to add these barcodes to list x and y and wants to remove it from a,b and c

Reply:
That's what I suggested you to be agnostic (or stupid like Unix): just don't care about what is going on. Just do one simple thing right.

Like
* you have a list of user lists, and a list of pre-checked user lists
* when the user finally clicks "OK" on the dialog, you return the list of selected user lists. You're done with your code.
* the caller compares the user lists, the initially pre-checked user lists and the returned selected user lists, and can see which lists were checked / unchecked.
* I'm afraid that if you're getting too specific inside your dialog code, you'll have to reinvent the wheel each time we need a user lists dialog.